### DISCUSSION TITLE
Add more docs about the real purpose of this plugin?

### DISCUSSION THREAD

**Discussion by folsze** [capawesome-team/capacitor-firebase#716] [General]
All I can find is:

🦋 Consistent versioning (no more SDK versions conflicts)
👁 Unified Typescript definitions
📄 Full documentation

Is this all the benefits this library provides?

Maybe some more info would be good, like what exactly should convince me to use this abstraction instead of going with the bare web-js sdk or the angular/fire library?

Probably the initial setup is easier, since this plugin adds configs to the native app folders. But other than that, is there benefits? Would be interesting to find out

**Comment by robingenz** [comment]
The biggest advantage is that you use the native SDKs. 

A few examples: 
- **App Check**: Support for App Attest and Play Integrity.
- **Authentication**: On Android, the user can log in to Google via a native popup with one click instead of opening an in-app browser where they first have to log in to Google using email/password before they can log in to your application.
- **Cloud Storage**: You do not have to load files into the WebView first, but can upload them directly from the native file system.
- **Cloud Firestore**: Reliable support for offline persistence (is not deleted so quickly).
- ...

**Comment by folsze** [comment]
Ok thanks for this answer. I guess I didn't think of the first few you mentioned, like the big Auth capabilities which I am using and which are super nice for me to use.

I failed to phrase my what I am wanting to know correctly. Really I was only looking for explanantion for firestore. I do not use other libraries but firestore & auth, so do not know abotu the other ones. 

But as a final feedback:
maybe it would be good to mention an overview what this plugin does for each firebase-thing. Like I could not find the benefits of firestore anywhere "Reliable support for offline persistence (is not deleted so quickly)." 

not in this main section:
https://github.com/capawesome-team/capacitor-firebase

nor in the specific section:
https://github.com/capawesome-team/capacitor-firebase/tree/main/packages/firestore

For auth it was clear what is adds for me since it was what I was looking for. But not for firestore. but maybe I did not read the docs  well enough. idk about other library parts, where a purpose docs would be good.

Thank you for this library again. I hope you can make use of my feedback somehow

**Reply by robingenz** [reply]
No, you are right. This is not yet in the documentation. If you use the Firebase Authentication plugin you **must** use the Cloud Firestore plugin if you want to use the Cloud Firestore security rules. Otherwise it would not be possible to identify the user during access. This is another important point.

**Reply by robingenz** [reply]
Thanks for your feedback!
